src/SignalRServiceExtension/Config/SignalRConfigProvider.cs

```diff
@@ -61,11 +55,7 @@ public void Initialize(ExtensionConfigContext context)
                 logger.LogWarning($"Unsupported service transport type: {serviceTransportTypeStr}. Use default {options.AzureSignalRServiceTransportType} instead.");
             }
 
-            serviceManager = new ServiceManagerBuilder().WithOptions(o =>
-            {
-                o.ConnectionString = options.ConnectionString;
-                o.ServiceTransportType = options.AzureSignalRServiceTransportType;
-            }).Build();
+            var serviceManager = CreateServiceManager(options.ConnectionString);
```

Comment:
Seems you created service manager in two places (another one in GetAzureSignalRClient)

Reply:
If connection string defined in app setting, `CreateServiceManager` will return a `ServiceManager`
If NO connection string defined in app setting, `CreateServiceManager` will return `null` 
But every request will invoke `GetAzureSignalRClient` and try to update `ServiceManager`. Not a bug, but make others confused. So in line 58 `StaticServiceHubContextStore.ServiceHubContextStore = new ServiceHubContextStore(null, loggerFactory);` is enough.